pkg/wasm/convert.go

```diff
@@ -63,6 +64,35 @@ func MaybeConvertWasmExtensionConfig(resources []*any.Any, cache Cache) bool {
 	return sendNack.Load()
 }
 
+// MaybeConvertWasmExtensionConfigDelta converts any presence of module remote download to local file.
+// It downloads the Wasm module and stores the module locally in the file system.
+func MaybeConvertWasmExtensionConfigDelta(resources []*discovery.Resource, cache Cache) bool {
```

Comment:
Feels cleaner to generate an any slices from resources slice before calling convert fucntion. That way you don't need to dup, but just use the same function, like the other handlers.